src/main/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -82,9 +82,15 @@
     abstract public SamReader open(final File file);
 
     public SamReader open(final Path path) {
-        final SamInputResource r = SamInputResource.of(path, getPathWrapper());
+        return open(path, getPathWrapper(), Function.identity());
```

Comment:
It seems like we should use the wrapper returned by `getPathWrapper()` as the default for both the bam and its index, instead of just the bam?